.github/workflows/security-check.yml

```diff
@@ -0,0 +1,125 @@
+        fi
+        
+        # Check for any modifications to Makefile
+        if echo "$CHANGED_FILES" | grep -q -E '^Makefile$'; then
+          MODIFIED_FILES="${MODIFIED_FILES} Makefile,"
+        fi
+        
+        # Remove trailing comma if any
+        MODIFIED_FILES=$(echo "$MODIFIED_FILES" | sed 's/,$//g')
+        
+        # For testing, force the workflow to treat all PRs as security risks
+        # if [ -n "$MODIFIED_FILES" ]; then
+        echo "FOR TESTING: Simulating security risk detection"
+        echo "cancel_workflows=true" >> $GITHUB_OUTPUT
+        echo "modified_files=$MODIFIED_FILES" >> $GITHUB_OUTPUT
+        if [ -z "$MODIFIED_FILES" ]; then
+          # If no sensitive files were found, add a dummy entry for testing
+          echo "modified_files=SIMULATED sensitive file (for testing)" >> $GITHUB_OUTPUT
+        fi
+        # else
+        #   echo "No sensitive files were modified by external contributor"
+        #   echo "cancel_workflows=false" >> $GITHUB_OUTPUT
+        #   echo "modified_files=" >> $GITHUB_OUTPUT
+        # fi
+        
+    - name: Warn about canceled workflows
+      if: steps.check-files.outputs.cancel_workflows == 'true'
+      run: |
+        echo "::warning::External contribution modifies sensitive files. Subsequent workflows will be canceled for security review."
+
+    - name: Comment on PR about security concerns
+      if: steps.check-files.outputs.cancel_workflows == 'true'
+      uses: actions/github-script@v6
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        script: |
+          const modifiedFiles = '${{ steps.check-files.outputs.modified_files }}';
+          await github.rest.issues.createComment({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            issue_number: context.issue.number,
+            body: `## ⚠️ Security Review Required for External Contribution ⚠️
+            
+            [THIS IS A TEST RUN - SIMULATING SECURITY CHECK]
+            
+            Thank you for your contribution! 
+            
+            As you're not a repository collaborator, this PR requires a security review because it modifies sensitive files:
+            - ${modifiedFiles}
+            
+            These files can affect the project's build system, CI/CD workflows, or other security-sensitive areas. All other automated checks have been temporarily paused until a maintainer reviews these changes.
+            
+            **Note to maintainers:** Please review these changes carefully before allowing other workflows to proceed.`
+          });
+        
+  cancel-workflows:
+    name: Cancel subsequent workflows if needed
+    needs: security-check
+    if: needs.security-check.outputs.cancel_workflows == 'true'
+    runs-on: ubuntu-latest
+    steps:
+    - name: Cancel subsequent workflows
+      uses: styfle/cancel-workflow-action@01ce38bf961b4e243a6342cbade0dbc8ba3f0432 # v0.12.0
```

Comment:
It still could happen that a workflow has leaked sensitive data before being canceled. I'd make that the other workflows don't start until this one signals that they can start.